src/peft/utils/save_and_load.py

```diff
@@ -130,14 +131,27 @@ def get_peft_model_state_dict(
     ):
         warnings.warn("Setting `save_embedding_layers` to `True` as embedding layers found in `target_modules`.")
         save_embedding_layers = True
+    # check if the vocab size of the base model is different from the vocab size of the finetuned model
+    elif (
+        save_embedding_layers == "auto"
+        and hasattr(model, "config")
+        and config.base_model_name_or_path
+        and model.config.vocab_size != AutoConfig.from_pretrained(config.base_model_name_or_path).vocab_size
+    ):
```

Comment:
I wonder if we should be super defensive here, just in case:

```python
    elif save_embedding_layers == "auto":
        vocab_size = getattr(getattr(model, config, None), "vocab_size", None)
        model_id = getattr(config, "base_model_name_or_path", None)
        if vocab_size and model_id and (vocab_size != AutoConfig.from_pretrained(model_id).vocab_size):
            ...
```

Reply:
Done, Thank you for the suggestion, makes sense.